video_creation/screenshot_downloader.py

```diff
@@ -71,6 +72,20 @@ def get_screenshots_of_reddit_posts(reddit_object: dict, screenshot_num: int):
 
         context.add_cookies(cookies)  # load preference cookies
 
+        # Login to Reddit
+        print_substep("Logging in to Reddit...")
+        page = context.new_page()
+        page.goto("https://www.reddit.com/login", timeout=0)
+        page.set_viewport_size(ViewportSize(width=1920, height=1080))
+        page.wait_for_load_state()
+
+        page.locator('[name="username"]').fill(settings.config["reddit"]["creds"]["username"])
+        page.locator('[name="password"]').fill(settings.config["reddit"]["creds"]["password"])
+        page.locator("button:has-text('Log In')").click()
+
+        page.wait_for_load_state()  # Wait for page to fully load and add 5 seconds
+        page.wait_for_timeout(5000)
```

Comment:
Could be improved by looking for an element on the home page

Reply:
Yeah but this works fine and makes sure the user is always logged in